### PR TITLE
Timetable post

### PR DESCRIPTION
**Pull request by david-yz-liu**
From #404:

This pull request allows a user to post a schedule to Facebook.

Only the fall schedule can be posted right now. It should be easy to snap in functionality for the spring session.
No half hour courses are accounted for. The diagram file does not support this (yet).
The image size/font size is disappointing. Diagrams seems to only output 'em's, which ImageMagick doesn't play well with in its current execution.

The commits may seem a bit funny in this pull request, I split up the diagrams/image conversion merging and timetable posting into two separate pull requests. Once the previous pull request is merged in, this should hopefully become clearer.

### CODE REVIEW

File: hs/Diagram.hs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+module Diagram (renderTable) where
+
+import Diagrams.Prelude
+import Diagrams.Backend.SVG.CmdLine
+import Diagrams.Backend.SVG
+import Data.List
+
+textFontSize = fontSize (Global 20)
+
+timetableStrings = [
```
Comment (david-yz-liu) [inline hs/Diagram.hs]: You can delete this

File: js/common/graph_image.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+function getGraphImage() {
+    'use strict';
+
+    var img;
+    var str = '';
+
+    $('td').each(function (i, elem) {
+        if ($(elem).attr('class') !== 'timetable-time' &&
+            !($(elem).attr('id').indexOf('.') > -1) &&
```
Comment (david-yz-liu) [inline js/common/graph_image.js]: Just use === -1?

File: js/common/grid_image.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    $('td').each(function (i, elem) {
+        if ($(elem).attr("class") !== 'timetable-time' &&
+            !($(elem).attr("id").indexOf('.') > -1) &&
+            !($(elem).attr("id").indexOf('S') > -1)) {
+            str = str + $(elem).html() + '%0A';
+        }
+    });
+
+    $.ajax({
+        url: 'timetable-image?courses=' + str,
```
Comment (david-yz-liu) [inline js/common/grid_image.js]: use the `data` key instead of putting the query parameter directly in the URL

**Comment by Ian-Stewart-Binks** [comment]
The new merge conflicts are interesting. I will take care of these.

**Comment by Ian-Stewart-Binks** [comment]
@david-yz-liu Ready to merge.

**Pull request merged**